openlibrary/catalog/get_ia.py

```diff
@@ -196,49 +129,59 @@ def get_data(loc):
     return buf
 
 def get_from_archive(locator):
+    """
+    Gets a single binary MARC record from within an Archive.org
+    bulk MARC item - data only.
+
+    :param str locator: Locator ocaid/filename:offset:length
+    :rtype: str|None
+    :return: Binary MARC data
+    """
+    data, offset, length = get_from_archive_bulk(locator)
+    return data
+
+def get_from_archive_bulk(locator):
+    """
+    Gets a single binary MARC record from within an Archive.org
+    bulk MARC item, and return the offset and length of the next
+    item.
+    If offset or length are `None`, then there is no next record.
+
+    :param str locator: Locator ocaid/filename:offset:length
+    :rtype: (str|None, int|None, int|None)
+    :return: (Binary MARC data, Next record offset, Next record length)
+    """
     if locator.startswith('marc:'):
         locator = locator[5:]
     filename, offset, length = locator.split (":")
-    offset = int (offset)
-    length = int (length)
-
-    ia, rest = filename.split('/', 1)
-
-    for attempt in range(5):
-        try:
-            host, path = find_item(ia)
-            break
-        except socket.timeout:
-            if attempt == 4:
-                raise
-            print 'retry, attempt', attempt
+    offset = int(offset)
+    length = int(length)
 
     r0, r1 = offset, offset+length-1
-    url = 'http://' + host + path + '/' + rest
+    # get the next record's length in this request
+    r1 += 5
+    url = base + filename
 
     assert 0 < length < 100000
```

Comment:
Can this constant be abstracted into a variable? Is this meant to be a num character length? Is it arbitrary?